### PR TITLE
Revamping KitOps use case doc

### PR DESCRIPTION
**Pull request by bmicklea**
Rewrote the use cases documentation  to focus on the two main ways KitOps is used in organizations today.

Updated some wording on the ToC to make things clearer and more consistent.

Relates to #262

### CODE REVIEW

File: docs/src/docs/use-cases.md [hunk 1]
```diff
@@ -1,12 +1,40 @@
-# KitOps Workflow
+# How KitOps Is Used 🛠️
 
-KitOps is the market's only open source, standards-based packaging and versioning system designed for AI/ML projects. You can read more about why KitOps is a powerful collaboration solution for AI/ML projects in our [Why KitOps](./why-kitops.md) page.
+KitOps is the market's only open source, standards-based packaging and versioning system designed for AI/ML projects. Using the OCI standard allows KitOps to be painlessly adopted by any organization using containers and enterprise registries today.
 
-## How KitOps Gets Used 🛠️
+## Stage 1: Handoff From Development and Production 🤝
```
Comment (brett-hodges) [inline docs/src/docs/use-cases.md:5]: I would change this to either:
A) Stage 1: Handoff between Development and Production
or
B) Stage 1: Handoff From Development to Production

File: docs/src/docs/use-cases.md [hunk 2]
```diff
@@ -1,12 +1,40 @@
...
+Many organizations have AI teams build a [ModelKit](./modelkit/intro.md) for each version of the AI project that is going to staging, user acceptance testing (UAT), or production.
 
-Here’s how we used ModelKits along with the tools we already used in our data science, software development, and infrastructure management teams.
+This ensures that:
+* __Operations teams have all the assets and information they need__ in order to determine how to test, deploy, audit, and manage these new workloads
+* __Organizations are protected against vendor shifts__ in their MLOps and Serving Infrastructure domains (this also gives them negotiating leverage with vendors)
+* __AI versioned packages are held in the same enterprise registry__ as other production assets like containers
+* __Compliance teams have a catalogue of versioned models__ that can be used for EU AI Act or other regulatory reporting
```
Comment (brett-hodges) [inline docs/src/docs/use-cases.md:13]: Maybe include a link for "EU AI Act" (https://artificialintelligenceact.eu/)

File: docs/src/docs/use-cases.md [hunk 3]
```diff
@@ -1,12 +1,40 @@
...
+* __Everyone has a library of immutable and signed ModelKits__ for intellectual property, progress tracking, or other requirements
```
Comment (brett-hodges) [inline docs/src/docs/use-cases.md:14]: Do you want to add "auditing"?

File: docs/src/docs/use-cases.md [hunk 4]
```diff
@@ -1,12 +1,40 @@
...
+## Stage 2: Storage for all AI Project Versions 💾
+
+Organizations that are more mature in their handling of AI projects, or are subject to extra scrutiny or regulations extend the use of ModelKits to the development phase as well. This solves the currently scattered storage of artifacts.
+
+This ensures that:
+* The organization uses a standards-based storage (OCI 1.1 artifacts) for all their AI/ML project work
+* Data and AI teams who don't work closely together can share artifacts even if they're using different development tools
+* Development artifacts can't be accidentally or maliciously tampered with
+
+This can be a lightweight process that relies on automating the ModelKit packing via the Kit [CLI](./cli/cli-reference.md).
```
Comment (brett-hodges) [inline docs/src/docs/use-cases.md:33]: Consider:  "This can be a lightweight process automated with ModelKit packing via the Kit [CLI]"

**Review by brett-hodges — COMMENTED** [review]
I added some suggested revisions, but overall this looks fantastic!

**Comment by bmicklea** [comment]
Pushed commit to address @brett-hodges feedback

**Review by brett-hodges — APPROVED** [review]

**Pull request merged**